### PR TITLE
Marshall Houston - Scrabble

### PR DESCRIPTION
**Pull request by marshallhouston**
WIP on scoring multiple letters.

### CODE REVIEW

File: test/scrabble_test.rb [hunk 1]
```diff
@@ -6,8 +6,40 @@
...
+
+  def test_it_can_score_an_entire_word
+    game = Scrabble.new
+    expected = 8
+    actual = game.score("hello")
+
+    assert_equal expected, actual
+  end
+
+  def test_it_can_score_a_blank_string
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb:21]: I like that you broke out individual tests for edge cases!

File: test/scrabble_test.rb [hunk 2]
```diff
@@ -6,8 +6,40 @@
...
+
+  def test_it_can_score_letter_multipliers
+    game = Scrabble.new
+    expected = 9
+    actual = game.score_with_multipliers("hello", [1,2,1,1,1])
+
+    assert_equal expected, actual
+  end
+
 end
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb:45]: I'd like to see you add a few more tests to double-check functionality (e.g. words of different lengths using different letters)

File: lib/scrabble.rb [hunk 3]
```diff
@@ -1,9 +1,33 @@
 class Scrabble
 
-  def score(word)
-    1
+
+
+  def score(word = nil)
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:5]: This method is a bit long. I'd love to see you push yourself to break out any nested ifs/blocks/loops (in this case, the `#each` block perhaps) into their own method; this helps clean up your method, work towards SRP & improve encapsulation.

File: lib/scrabble.rb [hunk 4]
```diff
@@ -1,9 +1,33 @@
...
+    score = 0
+    if word != nil
+      individual_letters = word.split('')
+      individual_letters.each do |letter|
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:9]: I'd also love to see you implementing more advanced enums; this could be a good case for map as you're 'mapping' scores onto each letter (then you can sum at the end). There are many possible solutions though & glad you got it to work!

File: lib/scrabble.rb [hunk 5]
```diff
@@ -1,9 +1,33 @@
...
+        point_value = point_values[letter.upcase]
+        score = score.to_i + point_value.to_i
+      end
+    end
+    score.to_i
+  end
+
+  def score_with_multipliers(word = nil, letter_multipliers)
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:17]: Same comments from above apply :)

**Review by VictoriaVasys — COMMENTED** [review]
Nice Job Marshall. Push yourself to create more tests to drive more robust code & break out logic in your implementation (and make sure to test any extra methods you create). Way to commit throughout the hour!

File: lib/scrabble.rb [hunk 6]
```diff
@@ -1,9 +1,33 @@
 class Scrabble
 
-  def score(word)
-    1
+
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:3]: Be cognizant of spacing; not a huge deal, but it makes your code more readable.

File: lib/scrabble.rb [hunk 7]
```diff
@@ -1,9 +1,33 @@
...
+        score = score.to_i + point_value.to_i
+      end
+    end
+    score.to_i
+  end
+
+  def score_with_multipliers(word = nil, letter_multipliers)
+
+    score = 0
+      individual_letters = word.split('')
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:20]: Since this isn't code within an if/else or loop or enum, it doesn't need to be indented.
